### PR TITLE
Update CLI code for `name_or_id` renamed to `announce_set`

### PR DESCRIPTION
**Pull request by david-crespo**
This is on top of the `integration` branch. I actually don't know how this sort of thing is done in this repo, but in any case here is the fix that needs to be done when the latest schema changes are brought into main so the CLI will compile.

### CODE REVIEW

File: cli/src/cmd_net.rs [hunk 1]
```diff
@@ -325,7 +325,7 @@ impl AuthenticatedCmd for CmdBgpWithdraw {
     async fn run(&self, client: &Client) -> Result<()> {
```
Comment (elaine-oxide) [inline cli/src/cmd_net.rs:325]: (Minor comment)
line 308: "to" -> "from"

File: cli/src/cmd_net.rs [hunk 2]
```diff
@@ -273,7 +273,7 @@ impl AuthenticatedCmd for CmdBgpAnnounce {
...
```
Comment (elaine-oxide) [inline cli/src/cmd_net.rs:273]: line 250: "address lot" should be "announce set"

Test:

Initial state:
```
$ cat cloud-pool-lot2.json
{
    "name": "cloud-pool2",
    "description": "a lot for cloud communications2",
    "kind": "pool",
    "blocks": [
        {
            "first_address": "203.0.153.1",
            "last_address": "203.0.153.254"
        }
    ]
}

$ oxide system networking address-lot create \
    --json-body cloud-pool-lot2.json
{
  "blocks": [
    {
      "first_address": "203.0.153.1",
      "id": "f75b71ae-5e4f-4c51-b37c-182cae3a54f1",
      "last_address": "203.0.153.254"
    }
  ],
  "lot": {
    "description": "a lot for cloud communications2",
    "id": "8da1c5ea-b49d-45a1-bcdf-b8138c067dc0",
    "kind": "pool",
    "name": "cloud-pool2",
    "time_created": "2024-09-20T16:54:39.661517Z",
    "time_modified": "2024-09-20T16:54:39.661517Z"
  }
}

$ oxide system networking bgp announcement list --announce-set as65547-announce
[
  {
    "address_lot_block_id": "387b0ed7-5aa4-4017-9c88-7343c1285605",
    "announce_set_id": "387b0ed7-5aa4-4017-9c88-7343c1285605",
    "network": "198.51.100.0/24"
  },
  {
    "address_lot_block_id": "387b0ed7-5aa4-4017-9c88-7343c1285605",
    "announce_set_id": "387b0ed7-5aa4-4017-9c88-7343c1285605",
    "network": "203.0.113.0/24"
  }
```

Invoke command:
```
$ oxide system networking bgp announce \
    --announce-set as65547-announce \
    --address-lot cloud-pool2 \
    --prefix 203.0.153.0/24
```

Resulting state:
```
$ oxide system networking bgp announcement list --announce-set as65547-announce
[
  {
    "address_lot_block_id": "db12d711-1989-4cac-b556-6cbf097fa4a5",
    "announce_set_id": "387b0ed7-5aa4-4017-9c88-7343c1285605",
    "network": "198.51.100.0/24"
  },
  {
    "address_lot_block_id": "db12d711-1989-4cac-b556-6cbf097fa4a5",
    "announce_set_id": "387b0ed7-5aa4-4017-9c88-7343c1285605",
    "network": "203.0.113.0/24"
  },
  {
    "address_lot_block_id": "db12d711-1989-4cac-b556-6cbf097fa4a5",
    "announce_set_id": "387b0ed7-5aa4-4017-9c88-7343c1285605",
    "network": "203.0.153.0/24"
  }
]
```

File: cli/src/cmd_net.rs (hunk 1 above)
Comment (elaine-oxide) [inline cli/src/cmd_net.rs:325]: line 309: "address lot" should be "announce set", there is no address lot specified here.

Test (continuation of test for line 250, use the resulting state from there for initial state here):

Invoke command:
```
$ oxide system networking bgp withdraw \
    --announce-set as65547-announce \
    --prefix 203.0.153.0/24
```

Resulting state:
```
$ oxide system networking bgp announcement list --announce-set as65547-announce
[
  {
    "address_lot_block_id": "792720c3-373d-4b01-b854-2f7c19402467",
    "announce_set_id": "387b0ed7-5aa4-4017-9c88-7343c1285605",
    "network": "198.51.100.0/24"
  },
  {
    "address_lot_block_id": "792720c3-373d-4b01-b854-2f7c19402467",
    "announce_set_id": "387b0ed7-5aa4-4017-9c88-7343c1285605",
    "network": "203.0.113.0/24"
  }
]
```

**Comment by elaine-oxide** [comment]
Tested this on a4x2 with:
- omicron `bac635f68831c8a3792156fcfd5ce13bef61fa37`
- Oxide CLI version compiled from this branch with this commit
```
$ oxide version
Oxide CLI 0.7.0+20240821.0
Built from commit: 53fedd7294b39bfe7ceeedf3c40bbe40d0c11e7a 
Oxide API: 20241009.0
```

Items tested via Oxide CLI:
https://github.com/oxidecomputer/omicron/issues/6467
https://github.com/oxidecomputer/omicron/pull/6498
https://github.com/oxidecomputer/omicron/pull/6560
https://github.com/oxidecomputer/omicron/issues/6468
https://github.com/oxidecomputer/oxide.rs/pull/843 (this PR)

File: cli/src/cmd_net.rs (hunk 1 above)
Comment (elaine-oxide) [reply cli/src/cmd_net.rs:325]: For clarity, here is the help output:
```
$ oxide system networking bgp withdraw --help
Withdraw a prefix over BGP.

This command removes the provided prefix to the specified announce set.
The remove is performed as a read-modify-write on the specified address lot.

Usage: oxide system networking bgp withdraw [OPTIONS] --announce-set <ANNOUNCE_SET> --prefix <PREFIX>

Options:
      --announce-set <ANNOUNCE_SET>
          The announce set to withdraw from

      --prefix <PREFIX>
          The prefix to withdraw

  -h, --help
          Print help (see a summary with '-h')

Global Options:
      --profile <PROFILE>
          Configuration profile to use for commands
```

File: cli/src/cmd_net.rs (hunk 2 above)
Comment (elaine-oxide) [reply cli/src/cmd_net.rs:273]: For clarity, here is the help output:
```
$ oxide system networking bgp announce --help
Announce a prefix over BGP.

This command adds the provided prefix to the specified announce set. It is
required that the prefix be available in the given address lot. The add is
performed as a read-modify-write on the specified address lot.

Usage: oxide system networking bgp announce [OPTIONS] --announce-set <ANNOUNCE_SET> --address-lot <ADDRESS_LOT> --prefix <PREFIX>

Options:
      --announce-set <ANNOUNCE_SET>
          The announce set to announce from

      --address-lot <ADDRESS_LOT>
          The address lot to draw from

      --prefix <PREFIX>
          The prefix to announce

      --description <DESCRIPTION>
          [default: ]

  -h, --help
          Print help (see a summary with '-h')

Global Options:
      --profile <PROFILE>
          Configuration profile to use for commands
```

**Review by ahl — APPROVED** [review]

**Pull request merged**